### DISCUSSION TITLE
How to correctly implement the Celery Chain with multiple links

### DISCUSSION THREAD

**Discussion by Martin25699** [celery/celery#9216] [Q&A]
Need to combine one audio track with several videos, currently there is a problem that the audio track is loaded several times.
```
t1 = download_video.si(url=video1)
t2 = download_video.si(url=video2)
t3 = download_audio.si(url=audio1)

d1 = group(t1, t3) | merge_video_and_audio.s()
d2 = group(t2, t3) | merge_video_and_audio.s()

finish = group(d1,d2) | send_report.s()
finish.apply_async()
```
in the current example, task3 will be launched 2 times, but I need it to be executed only 1 time. How can this be implemented?

**Comment by thedrow** [comment]
I'm not sure. Maybe you can provide us with more context?
